include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -256,14 +298,49 @@ struct tuple<T1, T...>
 
     using tuple_type = ::std::tuple<T1, T...>;
 
+    template <::std::size_t I>
+    constexpr auto
+    get() & -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const& -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() && -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const&& -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
     tuple() = default;
     tuple(const tuple& other) = default;
+    tuple(tuple&& other) = default;
     template <typename U1, typename... U>
     tuple(const tuple<U1, U...>& other) : holder(other.holder.value), next(other.next)
     {
     }
 
-    tuple(const T1& _value, const T&... _next) : holder(_value), next(_next...) {}
```

Comment:
Probably we need to keep this constructor as well.

Reply:
Probably, yes. Are tests affected?

Reply:
I thought about that tonight. Seems like for us there is no difference because the constructor would effectively do the same.

Reply:
Because of `std::forward` below, correct?